test/app/core/utils.spec.coffee

```diff
@@ -746,3 +756,34 @@ describe 'Utility library', ->
         ]
         expect(utils.findNextLevel(levels, 5, needsPractice)).toEqual(6)
         expect(utils.findNextAssessmentForLevel(levels, 5, needsPractice)).toEqual(-1)
+      it 'returns correct next levels when rc rc* rcl p r', ->
```

Comment:
fyi @AndrewJakubowicz, I think this is the specific failing case Stephanie found during her testing.